### PR TITLE
Make the incompatibility message more explicit

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:

Discussion in https://nvda.groups.io/g/nvda/topic/clock_add_on_add_on_store/105313955?p=,,,100,0,0,0::recentpostdate/sticky,,,100,2,0,105313955,previd%3D1712173725527803825,nextid%3D1710680040743854194&previd=1712173725527803825&nextid=1710680040743854194

### Summary of the issue:

The current incompatibility message for too old add-ons in the add-on store's details is as follows:
> An updated version of this add-on is required. The minimum supported API version is now 2024.1. This add-on was last tested with 2023.1. You can enable this add-on at your own risk. 

The concept of API is not clear for everyone and it is not explained in NVDA's GUI, nor in the User Guide. Thus, the logic of this sentence explaining why the add-on is not compatible is not easily understandable.

### Description of user facing changes
Modify the incompatibility reason for too old add-ons as follows:
> An updated version of this add-on is required. This add-on was last tested with 2023.1. The current version of NVDA requires that the extension be tested with NVDA 2024.1 or higher. You can enable this add-on at your own risk. 

### Description of development approach
Just changed the corresponding string.
### Testing strategy:
Manual check of the string in the GUI.
### Known issues with pull request:
None.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by CyrilleB79** [comment]
Cc @XLTechie, @Adriani90 for opinion or English spelling.

Cc @britechguy at the origin of the discussion; does it read better?

**Comment by britechguy** [comment]
@CyrilleB79 

The updated phrasing is vastly superior to the original.  As you can tell from the NVDA Group topic you cited, started by me, even I didn't exactly get what it was trying to tell me and I will admit that I had no idea that it was machine generated, so I thought the comment had somehow come from the add-on's developer.

The revised phrasing should be comprehensible to end-users and coders alike.

**Comment by CyrilleB79** [comment]
Note according to [this discussion,](https://nvda.groups.io/g/nvda/topic/a_clarification_about_nvda_s/105312364?p=,,,100,0,0,0::recentpostdate/sticky,,,100,2,0,105312364,previd%3D1712173725527803825,nextid%3D1710680040743854194&previd=1712173725527803825&nextid=1710680040743854194) the following message needs a clarification:
> Your NVDA configuration contains add-ons that are incompatible with this version of NVDA. These add-ons will be disabled after installation. After installation, you will be able to manually re-enable these add-ons at your own risk. If you rely on these add-ons, please review the list to decide whether to continue with the installation.

Do you think that I should also reword this one, e.g. indicating that the user can either update the add-on to a compatible version (if one is available) or re-enable incompatible add-ons at their own risks?

Just let me know.

**Comment by britechguy** [comment]
My revised wording would be as follows:
----
NVDA has determined you are using certain add-ons that are incompatible with this new version of NVDA.  If you continue installing, these add-ons will be disabled after it completes.  After installation, you may use the "Enable (override incompatibility)" option of the Action menu in the Add-On Store to re-enable these add-ons to see if they may still work, and you do so at your own risk.  If any add-on identified as incompatible is critically important to you, we recommend that you postpone updating NVDA until such time as a compatible version is available or create a portable copy of your current NVDA, including copying your existing user configuration, as a fall-back prior to updating.
------

Yes, it's long, but I think it presents the options in a comprehensible way and gives options to prevent "getting stuck without."

**Comment by Adriani90** [comment]
Thanks for this improvement, I think we will also have to revisit the disclaimer message in the update installer.

But for this one, what about something like this?

"This add-on was last tested with 2023.1 while the current NVDA version requires the add-on to be tested with 2024.1 or higher.
Therefore this add-on is incompatible with this NVDA version.
You can enable this add-on at your own risk by overriding the incompatibility in the add-on store, but it is highly recommended to contact the add-on author requesting to update it accordingly."

**Comment by britechguy** [comment]
@Adriani90 

Definitely works for me.  These notifications are for the end user, and even I can sometimes forget that.  They need not include every blessed detail, but should cut to the chase, and this does.

Personally, I love it.

**Comment by CyrilleB79** [comment]
@Adriani90 and @britechguy, thanks for your participation.

@Adriani90 wrote:
> Thanks for this improvement, I think we will also have to revisit the disclaimer message in the update installer.
> 
> But for this one, what about something like this?
> 
> "This add-on was last tested with 2023.1 while the current NVDA version requires the add-on to be tested with 2024.1 or higher. Therefore this add-on is incompatible with this NVDA version. You can enable this add-on at your own risk by overriding the incompatibility in the add-on store, but it is highly recommended to contact the add-on author requesting to update it accordingly."

It is not clear to me if this proposal is one or two suggestions? What is it meant to replace? Which dialog, or when performing which task? Thanks.

**Comment by Adriani90** [comment]
This was my proposal for the incompatibility message in the add-on storeVon meinem iPhone gesendetAm 04.04.2024 um 16:29 schrieb Cyrille Bougot ***@***.***>:﻿
@Adriani90 and @britechguy, thanks for your participation.
@Adriani90 wrote:

Thanks for this improvement, I think we will also have to revisit the disclaimer message in the update installer.
But for this one, what about something like this?
"This add-on was last tested with 2023.1 while the current NVDA version requires the add-on to be tested with 2024.1 or higher. Therefore this add-on is incompatible with this NVDA version. You can enable this add-on at your own risk by overriding the incompatibility in the add-on store, but it is highly recommended to contact the add-on author requesting to update it accordingly."

It is not clear to me if this proposal is one or two suggestions? What is it meant to replace? Which dialog, or when performing which task? Thanks.

**Comment by CyrilleB79** [comment]
> This was my proposal for the incompatibility message in the add-on store

Do you mean an alternative suggestion to what I have already modified in this PR, i.e. the message appearing in the Other details fields when an incompatible add-on is selected in the list?

Sorry to ask for confirmation again, but there are also other messages regarding compatibility in the Add-on Store, e.g. when installing or activating an incompatible add-on.

**Comment by Adriani90** [comment]
I meant indeed as an alternative to what you proposed in this pr, but we could look also how to streamline the messages so that they are not completely different or at least contain similar information. But I guess this is out of scope for this PR right? Von meinem iPhone gesendetAm 04.04.2024 um 16:53 schrieb Cyrille Bougot ***@***.***>:﻿

This was my proposal for the incompatibility message in the add-on store

Do you mean an alternative suggestion to what I have already modified in this PR, i.e. the message appearing in the Other details fields when an incompatible add-on is selected in the list?
Sorry to ask for confirmation again, but there are also other messages regarding compatibility in the Add-on Store, e.g. when installing or activating an incompatible add-on.

File: source/addonStore/models/version.py [hunk 1]
```diff
@@ -122,8 +122,8 @@ def getIncompatibleReason(
 				# The addon relies on older, removed features of NVDA, an updated add-on is required.
 				# The placeholder will be replaced with Year.Major.Minor (e.g. 2019.1).
 				"An updated version of this add-on is required. "
-				"The minimum supported API version is now {nvdaVersion}. "
 				"This add-on was last tested with {lastTestedNVDAVersion}. "
+				"The current version of NVDA requires that the extension be tested with NVDA {nvdaVersion} or higher. "
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply source/addonStore/models/version.py:126]:
```suggestion
				"NVDA requires this add-on to be tested with NVDA {nvdaVersion} or higher. "
```

**Review by Qchristensen — APPROVED** [review]
This change looks good, I'm happy with Sean's latest suggestion ("NVDA requires this add-on to be tested with NVDA {nvdaVersion} or higher. ")

**Pull request merged**